### PR TITLE
New RequestScopeHelper class to track request scopes across threads

### PR DESCRIPTION
**Pull request by spericas**
New class in FT that isolates code related to saving and restoring request scopes. The new class `RequestScopeHelper` provides support for CDI, HK2 and multiple JAX-RS applications; it includes additional code to track injection managers across threads. New test that verifies issue reported in #2632. For more information, see https://github.com/oracle/helidon/issues/2632#issuecomment-796831904

### CODE REVIEW

File: microprofile/fault-tolerance/src/main/java/io/helidon/microprofile/faulttolerance/MethodInvoker.java [hunk 1]
```diff
@@ -315,15 +298,8 @@ public boolean cancel(boolean mayInterruptIfRunning) {
...
-            requestController = CDI.current().select(RequestContextController.class).get();
-            requestScope = CDI.current().select(RequestScope.class).get();
-            requestContext = requestScope.referenceCurrent();
-        } catch (Exception e) {
-            requestScope = null;
-            LOGGER.fine(() -> "Request context not active for method " + method
-                    + " on thread " + Thread.currentThread().getName());
-        }
+        requestScopeHelper = new RequestScopeHelper();
+        requestScopeHelper.saveScope();
```
Comment (tjquinno) [inline microprofile/fault-tolerance/src/main/java/io/helidon/microprofile/faulttolerance/MethodInvoker.java:302]: Would we ever legitimately instantiate a helper without also saving the scope right away? 

If not, maybe fold the save work into the constructor? Or make the constructor private and use a static factory method that instantiates the helper and then invokes its `saveScope` method?

That would not work if there are ever cases where a single helper instance is created and then reused through several `save` and `clear` cycles.

**Review by tjquinno — COMMENTED** [review]
Looks good with one small question.

File: microprofile/fault-tolerance/src/main/java/io/helidon/microprofile/faulttolerance/MethodInvoker.java (hunk 1 above)
Comment (spericas) [reply microprofile/fault-tolerance/src/main/java/io/helidon/microprofile/faulttolerance/MethodInvoker.java:302]: I don't have a use case for saving/clearing/saving yet, so it is a bit more generic than necessary. Having said that, I think it easier to read with an explicit call to `saveScope`.

Comment (tjquinno) [reply microprofile/fault-tolerance/src/main/java/io/helidon/microprofile/faulttolerance/MethodInvoker.java:302]: My concern was that correct usage requires me to remember to both instantiate the helper and invoke `saveScope` explicitly. This is internal so I'm not overly worried.

**Comment by romain-grecourt** [comment]
Can you rename this PR to something meaningful and use `Fixes #2632`. Otherwise it shows up in the git log and does mean much after a while.

**Review by tjquinno — APPROVED** [review]

**Pull request merged**